File: Forge/src/main/java/hantonik/atomic/core/block/entity/IAtomicBlockEntityLocation.java
package hantonik.atomic.core.block.entity;

import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.state.BlockState;

public interface IAtomicBlockEntityLocation {
    BlockState state();

    BlockPos pos();

    Level level();
}

File: Forge/src/main/kotlin/hantonik/atomic/core/block/entity/IAtomicBlockEntityLocation.kt
package hantonik.atomic.core.block.entity

import net.minecraft.core.BlockPos
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.state.BlockState

interface IAtomicBlockEntityLocation {
    fun state(): BlockState

    fun pos(): BlockPos

    fun level(): Level?
}